openstack/content-repo/templates/cronjob.yaml

```
@@ -0,0 +1,85 @@
+{{- range $repo, $config := .Values.repos }}
+{{- if not $config.disabled }}
+apiVersion: batch/v1
+kind: CronJob
+
+metadata:
+  name: content-repo-{{ $repo }}
+  namespace: content-repo
+  labels:
+    system: openstack
+    component: content-repo
+
+spec:
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: 3
+  failedJobsHistoryLimit: 3
+  schedule: "TZ={{ $.Values.timezone }} {{ $config.schedule | default "59 23 * * *" }}"
+  # TODO: switch to when we have k8s 1.27 and remove TZ from schedule
+  # timeZone: {{ $.Values.timezone }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            component: content-repo-{{ $repo }}
+            alert-tier: os
+            alert-service: swift
+          annotations:
+            checksum/check.bin: {{ include "content-repo/templates/bin-configmap.yaml" $ | sha256sum }}
+            checksum/check.etc: {{ include "content-repo/templates/configmap.yaml" $ | sha256sum }}
+            checksum/check.secrets: {{ include "content-repo/templates/secret.yaml" $ | sha256sum }}
+            checksum/check.secrets-certs: {{ include "content-repo/templates/secret-certs.yaml" $ | sha256sum }}
+        spec:
+          restartPolicy: OnFailure
+          volumes:
+          - name: bin
+            configMap:
+              name: bin-content-repo
+              defaultMode: 0755
+          - name: config
+            configMap:
+              name: swift-http-import
+          - name: secret
+            secret:
+              secretName: swift-http-import-certs
+          containers:
+          - name: swift-http-import
+            image: {{ $.Values.global.registry }}/swift-http-import:{{ include "image_version" $.Values }}
+            command:
+              - /sbin/tini
+            args:
+              - /bin-content-repo/exponential-backoff
+              - /usr/bin/swift-http-import
+              - /etc/http-import/config/{{ $repo }}.yaml
```

If I'm not mistaken (and please double-check me on this), Kubernetes should do retries with exponential backoff on failure anyway, so we can also get rid of our custom exponential-backoff script. And with that removed, we can also get rid of tini since swift-http-import does not fork. Since the swift-http-import binary is the default entrypoint, that would all lead to just this:

```suggestion
            args:
              - /etc/http-import/config/{{ $repo }}.yaml
```